Sources/TuistSupport/Extensions/AbsolutePath+Extras.swift

```diff
@@ -17,6 +17,29 @@ public enum GlobError: FatalError, Equatable {
     }
 }
 
+extension Array where Element == AbsolutePath {
+    /// Packages should be added as a whole folder not individually.
+    /// (e.g. bundled file formats recognized by the OS like .pages, .numbers...)
+    ///
+    /// Can we improve this implementation with a more efficient approach?
+    /// - Returns: List of clean `AbsolutePath`s
+    public func cleanPackages() -> [AbsolutePath] {
```

Comment:
could you give me a little more context of why this is needed? What's the user input (`files` in `CopyFilesAction`) that makes this necessary?

Reply:
This can have a better name, but not sure what to name it...

Given the input input:
```
/User/name/project/Templates/yellow.template/meta.json
/User/name/project/Templates/yellow.template/image1.png
/User/name/project/Templates/yellow.template/image2.png
/User/name/project/Templates/blue.template/meta.json
/User/name/project/Templates/blue.template/image1.png
/User/name/project/Templates/blue.template/image2.png
/User/name/project/Fonts/somefont.ttf
```

This is the output:
```
/User/name/project/Templates/yellow.template
/User/name/project/Templates/blue.template
/User/name/project/Fonts/somefont.ttf
```

Reply:
It will basically clean the list for packages.

Reply:
Couldn't we pass the second list instead when declaring the list of files? Or is it because you want to use glob wildcards?

Reply:
Yeah, I thought about using glob wildcards just like for resources